jet_variations.py

```diff
@@ -4,1227 +4,94 @@
+                )
+
+            if USE_JES_INDIVIDUAL:  # individual JES sources, not recommended
+                with defaults(shift_key=["jet_jes_shift", "jet_jes_sources", "btag_sf_variation"]):
+                    for JEC_source in [
+                        "AbsoluteStat",
+                        "AbsoluteScale",
+                        "AbsoluteMPFBias",
+                        "Fragmentation",
+                        "SinglePionECAL",
+                        "SinglePionHCAL",
+                        "FlavorQCD",
+                        "TimePtEta",
+                        "RelativeJEREC1",
+                        "RelativeJEREC2",
+                        "RelativeJERHF",
+                        "RelativePtBB",
+                        "RelativePtEC1",
+                        "RelativePtEC2",
+                        "RelativePtHF",
+                        "RelativeBal",
+                        "RelativeSample",
+                        "RelativeFSR",
+                        "RelativeStatFSR",
+                        "RelativeStatEC",
+                        "RelativeStatHF",
+                        "PileUpDataMC",
+                        "PileUpPtRef",
+                        "PileUpPtBB",
+                        "PileUpPtEC1",
+                        "PileUpPtEC2",
+                        "PileUpPtHF",
+                    ]:
+                        add_shift(
+                            name=f"jesUnc{JEC_source}",
+                            shift_map={
+                                "Up": [1, f'{{"{JEC_source}"}}', f"up_jes{JEC_source}"],
+                                "Down": [-1, f'{{"{JEC_source}"}}', f"down_jes{JEC_source}"],
+                            },
+                        )
+            else:  # jes reduced set of sources, recommended
+                with defaults(shift_key=["jet_jes_shift", "jet_jes_sources"]):
+                    for name, JES_source in [
+                        ("jesUncAbsolute", '{"SinglePionECAL", "SinglePionHCAL", "AbsoluteMPFBias", "AbsoluteScale", "Fragmentation", "PileUpDataMC", "RelativeFSR", "PileUpPtRef"}'),
+                        ("jesUncAbsoluteYear", '{"AbsoluteStat", "TimePtEta", "RelativeStatFSR"}'),
+                        ("jesUncFlavorQCD", '{"FlavorQCD"}'),
+                        ("jesUncBBEC1", '{"PileUpPtEC1", "PileUpPtBB", "RelativePtBB"}'),
+                        ("jesUncBBEC1Year", '{"RelativeJEREC1", "RelativePtEC1", "RelativeStatEC"}'),
+                        ("jesUncHF", '{"RelativePtHF", "PileUpPtHF", "RelativeJERHF"}'),
+                        ("jesUncHFYear", '{"RelativeStatHF"}'),
+                        ("jesUncEC2", '{"PileUpPtEC2"}'),
+                        ("jesUncEC2Year", '{"RelativeJEREC2", "RelativePtEC2"}'),
+                        ("jesUncRelativeBal", '{"RelativeBal"}'),
+                        ("jesUncRelativeSampleYear", '{"RelativeSample"}')
+                    ]:
```

Comment:
I recommend to switch to the `Regrouped_*` definitions as e.g. [here](https://github.com/KIT-CMS/XYHBBTauTauAnalysis-CROWN/blob/12410959bcaccde6b937250df34f237b673646d6/jet_variations.py#L149). 
Futher, also here the btagging scalefactor needs to be added as already for the individual variations.